src/backend/cuda/jit.cpp

```diff
@@ -270,8 +273,23 @@ std::vector<char> compileToPTX(const char *ker_name, string jit_ker)
     nvrtcProgram prog;
     size_t ptx_size;
     std::vector<char> ptx;
-    NVRTC_CHECK(nvrtcCreateProgram(&prog, jit_ker.c_str(), ker_name, 0, NULL, NULL));
-    NVRTC_CHECK(nvrtcCompileProgram(prog, 0, NULL));
+    NVRTC_CHECK(nvrtcCreateProgram(&prog, jit_ker.c_str(),
+                                   ker_name, 0, NULL, NULL));
+
+    auto dev = getDeviceProp(getActiveDeviceId());
+    array<char, 32> arch;
+    snprintf(arch.data(), 32, "--gpu-architecture=compute_%d%d",
```

Comment:
Can you change `32` to `arch.size()` ? May be even change the array size to be a bit bigger. The size of the string is fairly close..

Reply:
Good catch! Fixed